### PR TITLE
[NIS-13912]: Implemented business logic tests

### PR DESCRIPTION
**Pull request by westcrime**

### CODE REVIEW

File: Offices/Offices.Tests/DoctorServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+
+namespace Offices.Tests
+{
+    public class DoctorServiceTests
+    {
+        private readonly IMapper _mapper;
+
+        public DoctorServiceTests()
+        {
+            var configuration = new MapperConfiguration(cfg =>
```
Comment (Glipchik) [inline Offices/Offices.Tests/DoctorServiceTests.cs:19]: Creating profile for unit tests can be considered as an option, but let's leave unit tests fully abstract and mock mapper as well. We'll create mapper config for integration tests

File: Offices/Offices.Tests/DoctorServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+            {
+                cfg.AddProfile<ApplicationMappingProfile>();
+            });
+
+            _mapper = configuration.CreateMapper();
+        }
+
+        [Fact]
+        public async void GetDoctorById_Returns_Doctor()
```
Comment (Glipchik) [inline Offices/Offices.Tests/DoctorServiceTests.cs:28]: I prefer following convention for unit tests: methodName_initialConditions_whatShouldHappen. Example: IsApplicable_LoanStatusNotDefaulted_ReturnsFalse
From your name it's not obvious what happens

File: Offices/Offices.Tests/DoctorServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+                              {
+                                  Id = "1231231-1231-123",
+                                  FirstName = "John",
+                                  LastName = fixture.Create<string>(),
+                                  MiddleName = fixture.Create<string>(),
+                                  OfficeId = fixture.Create<string>(),
+                                  Status = fixture.Create<string>()
+                              });
+
+            var officeRepositoryMock = new Mock<IOfficeRepository>();
```
Comment (Glipchik) [inline Offices/Offices.Tests/DoctorServiceTests.cs:44]: This and setting up doctor service should be done in the constructor

File: Offices/Offices.Tests/DoctorServiceTests.cs [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+
+            var doctorService = new DoctorService(doctorRepositoryMock.Object, officeRepositoryMock.Object, _mapper);
+
+            // Act
+            var doctorModel = await doctorService.Get("1231231-1231-123", CancellationToken.None);
```
Comment (Glipchik) [inline Offices/Offices.Tests/DoctorServiceTests.cs:49]: Magic string, move it somewhere and give it a name

File: Offices/Offices.Tests/DoctorServiceTests.cs [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+            });
+
+            _mapper = configuration.CreateMapper();
+        }
+
+        [Fact]
+        public async void GetDoctorById_Returns_Doctor()
+        {
+            // Arrange
+            var fixture = new Fixture();
```
Comment (Glipchik) [inline Offices/Offices.Tests/DoctorServiceTests.cs:31]: Create an attribute for autofixture and get objects as parameters for your test method

File: Offices/Offices.Tests/DoctorServiceTests.cs [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+            (
+                FirstName: fixture.Create<string>(),
+                LastName: fixture.Create<string>(),
+                MiddleName: fixture.Create<string>(),
+                OfficeId: "1231231-1231-123",
+                Status: fixture.Create<string>()
+            );
+
+            // Act
+            var exception = await Record.ExceptionAsync(async () =>
```
Comment (Glipchik) [inline Offices/Offices.Tests/DoctorServiceTests.cs:86]: Looks strange; any other ideas how to get exception from a delegate?

**Review by Glipchik — CHANGES_REQUESTED** [review]

File: Offices/Offices.Tests/DoctorServiceTests.cs (hunk 1 above)
Comment (westcrime) [reply Offices/Offices.Tests/DoctorServiceTests.cs:19]: Fixed

File: Offices/Offices.Tests/DoctorServiceTests.cs (hunk 2 above)
Comment (westcrime) [reply Offices/Offices.Tests/DoctorServiceTests.cs:28]: Fixed

File: Offices/Offices.Tests/DoctorServiceTests.cs (hunk 3 above)
Comment (westcrime) [reply Offices/Offices.Tests/DoctorServiceTests.cs:44]: Fixed

File: Offices/Offices.Tests/DoctorServiceTests.cs (hunk 4 above)
Comment (westcrime) [reply Offices/Offices.Tests/DoctorServiceTests.cs:49]: Fixed

File: Offices/Offices.Tests/DoctorServiceTests.cs (hunk 5 above)
Comment (westcrime) [reply Offices/Offices.Tests/DoctorServiceTests.cs:31]: Discussed, made methods for creating instances

File: Offices/Offices.Tests/DoctorServiceTests.cs (hunk 6 above)
Comment (westcrime) [reply Offices/Offices.Tests/DoctorServiceTests.cs:86]: Fixed, added shouldly

File: Offices/Offices.Data/Repositories/OfficeRepository.cs [hunk 7]
```diff
@@ -29,7 +29,7 @@ public override async Task DeleteAsync(string id, CancellationToken cancellation
             var officeToDelete = await (await _collection.FindAsync(d => d.Id == id, cancellationToken: cancellationToken)).FirstOrDefaultAsync(cancellationToken);
             if (officeToDelete == null)
             {
-                throw new NotFoundException($"Doctor with id {id} not found!");
+                throw new NotFoundException($"Office with id {id} not found!");
```
[the remark this answers is not in the document]
Comment (westcrime) [reply Offices/Offices.Data/Repositories/OfficeRepository.cs]: Fixed

File: Offices/Offices.Tests/DoctorServiceTests.cs [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+        [AutoData]
+        public async void Create_RelatedOfficeExists_ShouldBeSuccess(CreateDoctorModel createDoctorModel)
+        {
+            // Arrange
+            var office = CreateOffice(createDoctorModel.OfficeId, IsActive: true);
+
+            _officeRepositoryMock.Setup(repo => repo.GetAsync(createDoctorModel.OfficeId, CancellationToken.None))
+                .ReturnsAsync(office);
+
+            var doctorEntity = _fixture.Create<Doctor>();
```
[the remark this answers is not in the document]
Comment (westcrime) [reply Offices/Offices.Tests/DoctorServiceTests.cs:45]: Fixed

**Review by Glipchik — APPROVED** [review]

**Pull request merged**